### DISCUSSION TITLE
Invisible Material for roofs for indoor simulation

### DISCUSSION THREAD

**Discussion by steffenitely** [NVlabs/sionna#532] [Q&A] [answered]
Hi everybody ,
I want to do some indoor simulations and wanted to give my roof an invisible material so i can have a better overview of my scene when rendered. As suggested in the discussion "Adding Clip_at to render for indoor scenarios #398" i used a transparent BSDF for my invisible material and also tried to use a principled BSDF, but both did not result in an invisible roof/object. Is there a solution for this?

**Comment by merlinND** [comment, marked as answer]
Hello @steffenitely,

What do you see when calling `scene.preview()` with a `clip_at` argument? If you don't see any clipping happening, are you sure that you picked the right value? (it will depend on the scale of your scene along the vertical axis)

If using `scene.render()`, are you sure that the roofs have a `null` BSDF in the XML file that you load? What does the rendering look like?

**Reply by steffenitely** [reply]
Hi @merlinND 
Thank you for the response! The clip_at argument works :) Though I was hoping i could use the null BSDF if i wanted more than one invisible wall or as in #398 when rendering the scene. The Transparent BSDF, when exported in blender to a XML file, turns into a Diffuse BSDF and the Principled  BSDF turns into a twosided BSDF in the XML file. ith that i still see a solid coloured object. Does the Transparent BSDF work for you in a rendered scene when exported from Blender? If yes, maybe there is a problem with my Mitsuba add-on in Blender.
Otherwise, I will dig around in the Mitsuba BSDFs documentation to see if there is a way to create an invisible object by manualy changing the XML file after exporting it from Blender. 
But im always open for more suggestions :)
Hope you have a good one!

EDIT: Changing the BSDF inside the XML to a null BSDF seems to work. Just like you suggested. Thanks again

**Reply by merlinND** [reply]
> Does the Transparent BSDF work for you in a rendered scene when exported from Blender?

Actually you're right, the Blender exporter does *not* produce a `null` BSDF for Transparent materials. I got confused by looking at this _importer_ code:

https://github.com/mitsuba-renderer/mitsuba-blender/blob/5a12adda2d0da3ccf3c125a54e36f7cf57988957/mitsuba-blender/io/importer/materials.py#L549-L551

The exporter doesn't map Transparent to `null`, probably because a tint can be added, which is not supported by `null`.
